irispy/sji.py

```diff
@@ -878,3 +879,69 @@ def dustbuster(mc):
         map.data = image_orig
 
     return mc
+
+
+def read_iris_sji_level2_fits(filename):
+    """
+    Read IRIS level 2 SJI FITS from an OBS into an NDCube instance
+
+    Parameters
+    ----------
+    filename : `str`
+        File name to be read
+
+    Returns
+    -------
+    result: 'ndcube.NDCube'
+
+    """
+
+    # Open a fits file
+    my_file = fits.open(filename)
+    # Derive WCS, data and mask for NDCube from fits file.
+    wcs = WCS(my_file[0].header)
+    data = my_file[0].data
+    data_nan_masked = my_file[0].data
+    data_nan_masked[data == BAD_PIXEL_VALUE] = np.nan
+    mask = data_nan_masked == BAD_PIXEL_VALUE
+    # Derive unit and readout noise from detector.
+    exposure_times = my_file[1].data[:, my_file[1].header["EXPTIMES"]]
+    unit = iris_tools.DN_UNIT["SJI"]
+    readout_noise = iris_tools.READOUT_NOISE["SJI"]
+    # Derive uncertainty of data for NDCube from fits file.
+    uncertainty = u.Quantity(np.sqrt((data_nan_masked*unit).to(u.photon).value
+                                     + readout_noise.to(u.photon).value**2),
+                             unit=u.photon).to(unit).value
+    # Derive extra coordinates for NDCube from fits file.
+    times = np.array([parse_time(my_file[0].header["STARTOBS"])
+                      + timedelta(seconds=s)
+                      for s in my_file[1].data[:, my_file[1].header["TIME"]]])
+    pztx = my_file[1].data[:, my_file[1].header["PZTX"]] * u.arcsec
+    pzty = my_file[1].data[:, my_file[1].header["PZTY"]] * u.arcsec
+    xcenix = my_file[1].data[:, my_file[1].header["XCENIX"]] * u.arcsec
+    ycenix = my_file[1].data[:, my_file[1].header["YCENIX"]] * u.arcsec
+    obs_vrix = my_file[1].data[:, my_file[1].header["OBS_VRIX"]] * u.m/u.s
+    ophaseix = my_file[1].data[:, my_file[1].header["OPHASEIX"]]
```

Comment:
Have you confirmed that all these properties are in SJI FITS files?

Reply:
I am not at my office but I saw on tutorial that SJI headers have these properties. For now, I can say yes but I will try to go to office next week (hollidays) to check and to be 100 percent sure of that.

Reply:
Sorry to have made you wait last week. I certify that these 4 datas are in the my_file[1].header with the same name.